### PR TITLE
Update instruments profiling instructions

### LINKED ISSUE #24331: Update instructions for profiling using Instruments
## Reason
`instruments` command-line tool has been removed in [Xcode 13](https://developer.apple.com/documentation/xcode-release-notes/xcode-13-release-notes#Instruments) and has been replaced by `xctrace`. The instructions for profiling using Instruments are therefore outdated.

## Design
I will update the instructions using `xctrace` instead of `instruments`.

## Impact
Fixes instructions that don't work.

### PR DESCRIPTION
**Pull request by parikshitbajpai**
Closes #24331

### CODE REVIEW

**Comment by parikshitbajpai** [comment]
I can't check if the previous instructions still work on the older MacOS versions so I've left them as they were.

File: modules/doc/content/application_development/profiling.md [hunk 1]
```diff
@@ -268,7 +268,30 @@ Follow the steps below to get profiling information for your application:
...
+denotes a number, which you can open using
+
+```
+open Launch_mooseapp-oprof[0-9]+.trace
+```
+
+The Instruments application will open in a new window with the profile.
+The performance profile can be converted to [pprof](https://github.com/google/pprof)
+using [InstrumentsToPprof](https://github.com/google/instrumentsToPprof). To do this, open the trace in Instruments
+and select the processes you want to be converted, then copy the data using _Deep Copy_ in the _Edit_ menu. The copied
```
Comment (cticenhour) [inline modules/doc/content/application_development/profiling.md:287]:
```suggestion
and select the processes you want to be converted, then copy the data using *Deep Copy* in the *Edit* menu. The copied
```

"MooseDown" doesn't recognize the underscores for italics, so let's switch these to asterisks.

File: modules/doc/content/application_development/profiling.md [hunk 2]
```diff
@@ -268,7 +268,30 @@ Follow the steps below to get profiling information for your application:
...
+This will create a directory `Launch_mooseapp-oprof[0-9]+.trace`, where `[0-9]+`
+denotes a number, which you can open using
+
+```
+open Launch_mooseapp-oprof[0-9]+.trace
+```
+
+The Instruments application will open in a new window with the profile.
+The performance profile can be converted to [pprof](https://github.com/google/pprof)
+using [InstrumentsToPprof](https://github.com/google/instrumentsToPprof). To do this, open the trace in Instruments
```
Comment (cticenhour) [inline modules/doc/content/application_development/profiling.md:286]:
```suggestion
using [instrumentsToPprof](https://github.com/google/instrumentsToPprof). To do this, open the trace in Instruments
```

File: modules/doc/content/application_development/profiling.md [hunk 3]
```diff
@@ -268,7 +268,30 @@ Follow the steps below to get profiling information for your application:
...
+and select the processes you want to be converted, then copy the data using _Deep Copy_ in the _Edit_ menu. The copied
+data can be piped to `instrumentsToPprof` as
+```
+pbpaste | instrumentsToPprof
+```
+which will produce a `profile.pb.gz` which can be analyzed using `pprof` following the same instructions as above.
```
Comment (cticenhour) [inline modules/doc/content/application_development/profiling.md:292]:
```suggestion
data can be piped to `instrumentsToPprof` as

```
pbpaste | instrumentsToPprof
```

which will produce a `profile.pb.gz` which can be analyzed using `pprof` following the same instructions as above.
```

Need to add some space for this to properly render as a code snippet block.

**Review by cticenhour — CHANGES_REQUESTED** [review]
Putting in an initial review of the markdown content, with a double-check of the new directions with a MOOSE app likely occurring tomorrow (maybe today if I have time). 

Thanks for updating this! Just a few changes to improve the website render.

File: modules/doc/content/application_development/profiling.md [hunk 4]
```diff
@@ -268,7 +268,30 @@ Follow the steps below to get profiling information for your application:
...
+
+```
+open Launch_mooseapp-oprof[0-9]+.trace
+```
+
+The Instruments application will open in a new window with the profile.
+The performance profile can be converted to [pprof](https://github.com/google/pprof)
+using [InstrumentsToPprof](https://github.com/google/instrumentsToPprof). To do this, open the trace in Instruments
+and select the processes you want to be converted, then copy the data using _Deep Copy_ in the _Edit_ menu. The copied
+data can be piped to `instrumentsToPprof` as
```
Comment (cticenhour) [inline modules/doc/content/application_development/profiling.md:288]:
```suggestion
data can be piped to `instrumentsToPprof` as

```

Had to split this suggestion into two....GitHub markdown and our markdown had a bit of a conflict...

File: modules/doc/content/application_development/profiling.md (hunk 3 above)
Comment (cticenhour) [inline modules/doc/content/application_development/profiling.md:292]:
```suggestion

which will produce a `profile.pb.gz` which can be analyzed using `pprof` following the same instructions as above.
```

**Review by cticenhour — APPROVED** [review]
Tested the instructions this morning - works great on Ventura 13.3.1. Pending the last test recipe passing (had to invalidate due to a bad HTTPS gateway on CIVET), this is good to go.

**Pull request merged**